jobs/client-regeneration/client_regeneration/main.py

```diff
@@ -262,13 +266,121 @@ def write_baseline_clients_daily(client, seed, start_date):
   FROM numbered
   WHERE rn = 1
   """
-  job = client.query(q)
-  job.result()
-  return(table_name)
+    job = client.query(q)
+    job.result()
+    return table_name
+
+
+def init_baseline_clients_yearly(client, seed):
+    clients_yearly_name = f"mozdata.analysis.regen_sim_replaced_clients_yearly_{seed}"
+    clients_daily_name = f"mozdata.analysis.regen_sim_replaced_baseline_clients_daily_with_searches_{seed}"
```

Comment:
looks like the `_with_searches` query isn't in this script yet. we should probably add it since an upstream dependency of clients_yearly 
```
def write_baseline_clients_daily_with_searches(seed, start_date):
  table_name = """mozdata.analysis.regen_sim_replaced_baseline_clients_daily_with_searches_{}""".format(str(seed))
  q = f"""
  CREATE OR REPLACE TABLE {table_name}
  AS
  WITH
  base AS (
      SELECT
        c.client_id,
        c.first_seen_date AS first_seen_date,
        c.submission_date,
        c.country,
        c.device_model,
        COALESCE(a.searches, 0) AS searches,
        COALESCE(a.searches_with_ads, 0) AS searches_with_ads,
        COALESCE(a.ad_clicks, 0) AS ad_clicks,
      FROM `mozdata.fenix.baseline_clients_daily` c
      LEFT JOIN `mozdata.fenix.attributable_clients_v2` a
      USING(client_id, submission_date)
      WHERE c.submission_date >= DATE("{str(start_date)}")
  ),

  replaced AS (
      SELECT
        COALESCE(r.replacement_id, c.client_id) AS client_id,
        COALESCE(r.first_seen_date, c.first_seen_date) AS first_seen_date,
        udf.safe_sample_id(COALESCE(r.replacement_id, c.client_id)) AS sample_id,
        c.country,
        c.device_model,
        c.submission_date,
        ARRAY_AGG(c.client_id IGNORE NULLS ORDER BY regened_last_date) AS original_client_id,
        ARRAY_AGG(c.first_seen_date IGNORE NULLS ORDER BY regened_last_date) AS original_first_seen_date,
        SUM(searches) AS searches,
        SUM(searches_with_ads) AS searches_with_ads,
        SUM(ad_clicks) AS ad_clicks,
      FROM base c
      LEFT JOIN `mozdata.analysis.regen_sim_client_replacements_{str(seed)}` r
      -- we want the history starting on the regen date.
      ON (c.client_id = r.client_id) AND (c.submission_date BETWEEN r.regen_date AND r.regened_last_date)
      GROUP BY 1,2,3,4,5,6
  )

  SELECT
      *
  FROM replaced
  """

  job = client.query(q)
  job.result()
  return(table_name)
```

Reply:
@irrationalagent Should this (and `baseline_clients_daily`) use an `end_date` as well?

Reply:
Yea, OK, maybe we should do it for consistency and efficiency. Without an end date, I believe the the resulting table will be identical to the non-replaced source tables after last `regened_last_date` in the replacement mappings. I guess it would also help reduce confusion about us only meaning to replace pre-bugfix data.